src/java/org/apache/cassandra/transport/messages/BatchMessage.java

```diff
@@ -237,6 +239,11 @@ public BatchMessage(BatchStatement.Type type, List<Object> queryOrIdList, List<L
             {
                 FullQueryLogger.instance.logBatch(batchType.name(), queryStrings, values, options, fqlTime);
             }
+            if(auditLogEnabled)
```

Comment:
Here when someone enables audit log then every single query will be audited. In my opinion this should be made flexible by allowing percentage of queries to be audited as some of high throughput system may not be able to tolerate penalty of auditing for every single query.

Reply:
Ideally, AuditLog would not be percentage based but it would be every statement to meet the stated AuditLog requirements. For data sampling, we might not want to overload AuditLog, since it deviates the actual purpose of AuditLog. However, we can create separate JIRA to start this discussion if you would like.